src/editor/components/field/QueryEditorFieldSection.tsx

```diff
@@ -40,11 +32,18 @@ export const QueryEditorFieldSection = (config: FieldSectionConfiguration): Reac
         <QueryEditorField
           fields={props.fields}
           templateVariableOptions={props.templateVariableOptions}
-          onChange={onChange}
+          onChange={onChange(propsOnChange)}
           value={expression.property}
         />
         {props.children}
       </QueryEditorSection>
     );
   };
 };
+
+const onChange = (propsOnChange: (value: QueryEditorExpression) => void) => (property: QueryEditorProperty) => {
+  propsOnChange({
+    type: QueryEditorExpressionType.Property,
+    property,
+  } as QueryEditorPropertyExpression);
+};
```

Comment:
The only way I figured that we could solve the eslint error with was to extract a proper function

Reply:
Feels like a good solution! The other one is only an optimization that can be skipped for now.